src/components/Layout/Header/NavBar/UserMenu.tsx

```diff
@@ -98,7 +98,15 @@ const WalletButton: FC<WalletButtonProps> = ({
       isLoading={isLoadingLocalWallet}
       leftIcon={
         <HStack>
-          {isConnected ? undefined : <WarningTwoIcon ml={2} w={3} h={3} color='yellow.500' />}
+          {isConnected || walletInfo?.deviceId === 'DemoWallet' ? undefined : (
```

Comment:
An observation: I'm not sure which is correct, but the existing pattern in the codebase is to return `null`, rather than `undefined`, from a conditional in JSX.

That said, I can't find any documentation asserting that `null` is the "right" way to do it, and `undefined` feels more right to me.

Reply:
By **convention**, short-circuiting JSX renders in React with ternaries is done with `null`, with the happy render path first: 
```suggestion
          {!isConnected || walletInfo?.deviceId !== 'DemoWallet' ? happyJsxPath : null
```

In theory, react handles `false`, `null` and `undefined` the same way however: https://reactjs.org/docs/jsx-in-depth.html#booleans-null-and-undefined-are-ignored
> false, null, undefined, and true are valid children. They simply don’t render.

It means that this can be simplified with an expression such as (pseudo-code written in the comment box, obviously untested):

```suggestion
          {(!isConnected || walletInfo?.deviceId !== 'DemoWallet') ?? happyJsxPath}
```

Reply:
TIL @gomesalexandre  👍